### PR TITLE
Fix service bugs

### LINKED ISSUE #244: [PE-D] Unexpected behaviour due to clearsvc command
The clearsvc command works well in clearing all the services. However, it leads to alot of unexpected behaviour. I added a new service (dye hair - the new SC000) to the empty list of services. But when I use the command "findapt s/SC000", 2 results were shown - service of lash lift (this was the sc000 before clearsvc was called) and service for dye hair. The team may wish to relook at this issue on the existence of old and invalid services in appointments list. 

-------------
Labels: `severity.Medium` `type.FeatureFlaw`
original: chiamyunqing/ped#7

### PR DESCRIPTION
**Pull request by hanstirtaputra**
closes #244 

Changed `serviceCodeGeneration` method to take into account the current list of all appointments in Homerce, so as to prevent a duplicate `ServiceCode` from being generated after a service has been deleted and added.

`clearsvc` command now checks if there are any existing services that have been scheduled for future appointments. If so, then `clearsvc` will not execute. The user will have to delete the upcoming appointments with the service before he/she can `clearsvc`. This behavior is the same for `deletesvc` which has been implemented already.

### CODE REVIEW

File: src/main/java/seedu/homerce/logic/commands/service/ClearServiceCommand.java [hunk 1]
```diff
@@ -18,9 +25,25 @@
 
 
     @Override
-    public CommandResult execute(Model model, HistoryManager historyManager) {
+    public CommandResult execute(Model model, HistoryManager historyManager) throws CommandException {
         requireNonNull(model);
+
+        model.updateFilteredServiceList(PREDICATE_SHOW_ALL_SERVICES);
+        model.updateFilteredAppointmentList(PREDICATE_SHOW_ALL_APPOINTMENTS);
```
Comment (khoongwk) [inline src/main/java/seedu/homerce/logic/commands/service/ClearServiceCommand.java]: When we call update, the UI will change if the user is on the appointments tab. Not sure if this is an acceptable outcome.

**Review by khoongwk — APPROVED** [review]

**Review by khoongwk — APPROVED** [review]
LGTM

File: src/main/java/seedu/homerce/logic/commands/service/ClearServiceCommand.java (hunk 1 above)
Comment (hanstirtaputra) [reply src/main/java/seedu/homerce/logic/commands/service/ClearServiceCommand.java]: Thanks for pointing that out! I have changed the implementation to fix this.

**Review by RuiFengg — APPROVED** [review]
LGTM

**Pull request merged**